src/librustc_mir/transform/uneval_const_set.rs

```diff
@@ -0,0 +1,40 @@
+use crate::transform::{MirPass, MirSource};
+use rustc_middle::mir::visit::MutVisitor;
+use rustc_middle::mir::{BodyAndCache, Constant, Location};
+use rustc_middle::ty::{ConstKind, TyCtxt};
+
+pub struct UnevalConstSet;
+
+impl<'tcx> MirPass<'tcx> for UnevalConstSet {
+    fn run_pass(&self, tcx: TyCtxt<'tcx>, _: MirSource<'tcx>, body: &mut BodyAndCache<'tcx>) {
+        assert!(body.uneval_consts.is_empty());
+        let mut uneval_consts = Vec::new();
+        UnevalConstSetVisitor::new(tcx, &mut uneval_consts).visit_body(body);
+        body.uneval_consts = uneval_consts;
+    }
+}
+
+struct UnevalConstSetVisitor<'a, 'tcx> {
+    tcx: TyCtxt<'tcx>,
+    uneval_consts: &'a mut Vec<Constant<'tcx>>,
+}
+
+impl<'a, 'tcx> UnevalConstSetVisitor<'a, 'tcx> {
+    pub fn new(tcx: TyCtxt<'tcx>, uneval_consts: &'a mut Vec<Constant<'tcx>>) -> Self {
+        UnevalConstSetVisitor { tcx, uneval_consts }
+    }
+}
```

Comment:
Seems unnecessary; I'd just inline the construction here since it's used once.

Reply:
Do you meant me to remove the function and move the construction of the visitor above?. I can do that, it's more or less the same to me :).

Reply:
Yes, that. :)

Reply:
Done